src/Couchbase.Lite.Shared/API/Database/Database.cs

```diff
@@ -971,20 +971,17 @@ public void SaveBlob(Blob blob)
         /// <exception cref="ArgumentException">Throw if the given blob dictionary is not valid.</exception>
         /// <returns>The contained value, or <c>null</c> if it's digest information doesn’t exist.</returns>
         [CanBeNull]
-        public Blob GetBlob(Dictionary<string, object> BlobDict)
+        public Blob GetBlob(Dictionary<string, object> blobDict)
         {
-            if (!BlobDict.ContainsKey(Blob.DigestKey) || BlobDict[Blob.DigestKey] == null)
+            if (!blobDict.ContainsKey(Blob.DigestKey) || blobDict[Blob.DigestKey] == null)
                 return null;
 
-            if (!BlobDict.ContainsKey(Constants.ObjectTypeProperty) || (string) BlobDict[Constants.ObjectTypeProperty] != Constants.ObjectTypeBlob
-                || (BlobDict.ContainsKey(Blob.ContentTypeKey) && BlobDict[Blob.ContentTypeKey].GetType() != typeof(string))
-                || (BlobDict.ContainsKey(Blob.LengthKey) && BlobDict[Blob.LengthKey].GetType() != typeof(int))
-                || BlobDict[Blob.DigestKey].GetType() != typeof(string)) {
+            if (!Blob.IsBlob(blobDict)) {
                 throw new ArgumentException(CouchbaseLiteErrorMessage.InvalidJSONDictionaryForBlob);
             }
 
             C4BlobKey expectedKey = new C4BlobKey();
-            var keyFromStr = Native.c4blob_keyFromString((string)BlobDict[Blob.DigestKey], &expectedKey);
+            var keyFromStr = Native.c4blob_keyFromString((string)blobDict[Blob.DigestKey], &expectedKey);
```

Comment:
is this only for validating the size? since this is a lite core API to check for size and not mentioned in API spec. I am wondering is this necessary! 

@pasin  your thoughts?

Reply:
In line 976, I have the logic checking that. Since we still allow blob to be created without a digest, so I am not sure if I should add that logic in IsBlob logic.....

Reply:
What is the purpose of getting key from string here?

Reply:
We have a test case in GetBlob to test with digest with a random string as value. We need to use the random string to get the C4Blob key and then use the key we get the check if c4blob_getSize returns a valid value, otherwise null if -1 is returned.

Reply:
See, I didn't see get size for some reason. Thanks.